Stratis.Bitcoin/BlockPulling/LookaheadBlockPuller.cs

```diff
@@ -190,7 +196,8 @@ public LookaheadBlockPuller(ConcurrentChain chain, IConnectionManager connection
         public void SetLocation(ChainedBlock tip)
         {
             Guard.NotNull(tip, nameof(tip));
-            this.location = tip;
+            lock (this.locationLock)
```

Comment:
And I agree with you, I was just testing the grounds. I don't like it either. Will fix.